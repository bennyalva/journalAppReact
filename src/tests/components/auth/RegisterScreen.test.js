import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';
const { mount } = require("enzyme");
const { RegisterScreen } = require("../../../components/auth/RegisterScreen");

const middlewares = [thunk]; // config store
const mockStore = configureStore(middlewares); // config store

const initState = { // inicia es estado del store que se necesita cuando se renderiza el componente
    auth: {},
    ui: {
        loading: false,
        msgError: ''
    }
};

let store  = mockStore(initState);
// store.dispatch = jest.fn(); // para manejar los dispatch del store pero solo para las que son asyncronas(Promise)


describe('Test onRegisterScreen', () => {
    const wrapper = mount(
    <Provider store={ store }>
       <MemoryRouter>
       <RegisterScreen />
       </MemoryRouter>
    </Provider>)
   test('should call show correctly', () => {
       expect(wrapper).toMatchSnapshot()
   });

   test('should do dispatch acciÃ³n respective', () => {
        const emailField = wrapper.find('input[name="email"]');
        emailField.simulate('change', {
            target: {
                value: '',
                name: 'email'
            }
        });
        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });
        const actions = store.getActions();
        expect(actions[0].type).toEqual( types.uiSetError);
   });
   test('should show alert box', () => {
    const initState = { // inicia es estado del store que se necesita cuando se renderiza el componente
        auth: {},
        ui: {
            loading: false,
            msgError: 'Email no es correcto'
        }
    };
    
    const store  = mockStore(initState);
    const wrapper = mount(
        <Provider store={ store }>
           <MemoryRouter>
           <RegisterScreen />
           </MemoryRouter>
        </Provider>);

        expect(wrapper.find('.auth__alert-error').exists()).toBeTruthy()
   });
   
   
    
});
