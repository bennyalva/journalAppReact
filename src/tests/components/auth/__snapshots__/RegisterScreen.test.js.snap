// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test onRegisterScreen should call show correctly 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate_faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="benny"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="benny@gmail.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="Password"
      value="123456"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="passeord2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="text"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-1"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already register ?
    </a>
  </form>,
]
`;
